src/renderer/components/kube-object-menu/kube-object-menu.test.tsx

```diff
@@ -0,0 +1,295 @@
+
+describe("kube-object-menu", () => {
+  let hideDetailsStub: () => void;
+  let editResourceTabStub: () => void;
+  let apiManagerStub: IGettableStore;
+  let kubeObjectMenuRegistry: KubeObjectMenuRegistry;
+  let objectStub: KubeObject | null;
+  let dependencies: KubeObjectMenuDependencies<KubeObject>;
+
+  beforeEach(() => {
+    // TODO: Remove illegal global overwrites for what should be a dependency somewhere.
+    // TODO: Remove usage of experimental browser API.
+    window.requestIdleCallback = (callback: IdleRequestCallback): number => {
+      callback(undefined);
+
+      return undefined;
+    };
+
+    window.cancelIdleCallback = () => {};
+
+    apiManagerStub = {
+      getStore: <TKubeObjectStore extends KubeObjectStore<KubeObject>>(
+        // eslint-disable-next-line unused-imports/no-unused-vars-ts
+        api: string | KubeApi<KubeObject>,
+      ): TKubeObjectStore | undefined => undefined,
+    };
+
+    // TODO: Remove global shared state
+    KubeObjectMenuRegistry.resetInstance();
+    KubeObjectMenuRegistry.createInstance();
+    kubeObjectMenuRegistry = KubeObjectMenuRegistry.getInstance();
+
+    const MenuItemComponent: React.FC = () => <li>Some menu item</li>;
+
+    addDynamicMenuItem({
+      kubeObjectMenuRegistry,
+      MenuItemComponent,
+      apiVersions: ["some-api-version"],
+      kind: "some-kind",
+    });
+
+    addDynamicMenuItem({
+      kubeObjectMenuRegistry,
+      MenuItemComponent,
+      apiVersions: ["some-unrelated-api-version"],
+      kind: "some-kind",
+    });
+
+    addDynamicMenuItem({
+      kubeObjectMenuRegistry,
+      MenuItemComponent,
+      apiVersions: ["some-api-version"],
+      kind: "some-unrelated-kind",
+    });
+
+    hideDetailsStub = () => {};
+
+    editResourceTabStub = () => {};
+
+    dependencies = {
+      clusterName: "Some cluster name",
+      apiManager: apiManagerStub,
+      kubeObjectMenuRegistry,
+      hideDetails: hideDetailsStub,
+      editResourceTab: editResourceTabStub,
+    };
+  });
+
+  it("given no kube object, renders", () => {
+    objectStub = null;
+
+    const { baseElement } = render(
+      <KubeObjectMenu
+        object={objectStub}
+        toolbar={true}
+        dependencies={dependencies}
+      />,
+    );
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  describe("given kube object", () => {
+    let baseElement: Element;
+    let removeActionMock: any;
+    let getByTestId: (arg0: string) => any;
+    let queryByTestId: (arg0: string) => any;
+
+    beforeEach(async () => {
+      objectStub = KubeObject.create({
+        apiVersion: "some-api-version",
+        kind: "some-kind",
+        metadata: {
+          uid: "some-uid",
+          name: "some-name",
+          resourceVersion: "some-resource-version",
+          namespace: "some-namespace",
+        },
+      });
+
+      removeActionMock = asyncFn();
+
+      ({ baseElement, getByTestId, queryByTestId } = render(
```

Comment:
I think nowadays they recommend to use `screen` for queries (https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen) e.g. `screen.getByTestId`, so you probably don't need to have `let getByTestId` but better use `screen.getByTestId()` in each block.

Reply:
Good catch. Resolved